store/v2/commitment/store.go

```diff
@@ -53,7 +58,12 @@ func (c *CommitStore) WriteChangeset(cs *corestore.Changeset) error {
 		if !ok {
 			return fmt.Errorf("store key %s not found in multiTrees", key)
 		}
+		var prevKey []byte
 		for _, kv := range pairs.StateChanges {
+			if bytes.Compare(kv.Key, prevKey) < 1 {
```

Comment:
We shouldn't mandate that other SC trees suffer from this same limitation that IAVL does though right? In other words let's only make this check if the backing store is an IAVL tree.